android-components/components/feature/media/src/main/java/mozilla/components/feature/media/service/MediaSessionServiceDelegate.kt

```diff
@@ -180,10 +183,29 @@ internal class MediaSessionServiceDelegate(
     }
 
     @VisibleForTesting
+    @Suppress("TooGenericExceptionCaught")
     internal fun startForeground(sessionState: SessionState) {
         notificationScope?.launch {
             val notification = notificationHelper.create(sessionState, mediaSession)
-            service.startForeground(notificationId, notification)
+            try {
+                service.startForeground(notificationId, notification)
+            } catch (e: Exception) {
+                if (
+                    Build.VERSION.SDK_INT >= Build.VERSION_CODES.S &&
+                    e is ForegroundServiceStartNotAllowedException
+                ) {
+                    // We should not encounter this exception if `android:foregroundServiceType="mediaPlayback"`
+                    // is added to the service. The crash reporter loses the stack trace for this
+                    // exception so we want to be able to track this crash independently to ensure
+                    // this case is fixed and be able to determine if there are other cases where we
+                    // might be trying to start foreground services from the background.
+                    // https://bugzilla.mozilla.org/show_bug.cgi?id=1802620
+                    emitCaughtForegroundServiceStartNotAllowed(e)
```

Comment:
I don't have much context here, but just looking through I am wondering if it might make more sense to add a crash reporter as a (optional?) property to the `AbstractMediaSessionService` and inject it during instantiation like is done with the `BrowserStore`. It could then be passed down to this delegate as needed.

IMO that seems slightly more idiomatic, as AFAIK `Fact`s are intended to map more towards metrics we are recording with Glean.